### DISCUSSION TITLE
Nginx 502 Bad Gateway

### DISCUSSION THREAD

**Discussion by DannyJoris** [ddev/ddev#4784] [General]
I ran into a persistent nginx "502 Bad Gateway" error, seemingly out of the blue, which doesn't go away after restarting ddev.

`ddev logs` shows the following line:
```
2023/03/24 19:37:58 [error] 2123#2123: *349 upstream sent too big header while reading response header from upstream, client: 172.18.0.7, server: , request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm.sock:", host: "spine-health9.vh.ddev.site", referrer: "https://..."
```

It seems to suggest I should increase the `large_client_header_buffers` size as done here? https://github.com/ddev/ddev/pull/3034/files#diff-6480821958a08d0a2cb64353db2f99ccb410a30a8d27a7e4d28008b886fe69d0

I've added the following to `.ddev/nginx_full/nginx-site.conf`, but without success:
```
client_header_buffer_size 32k;
large_client_header_buffers 4 32k;
```

I commented out the code I was working on, cleared Drupal caches, but nothing seems to work. Any ideas?

**Comment by rfay** [comment]
Hi @DannyJoris - you did the right thing checking `ddev logs`. 

It appears that your PHP app is sending an absolutely huge header back to nginx. 

I suspect you did the right thing trying those options, but you may need to go far higher than that. And remember you have to `ddev restart after changing nginx-site.conf, and that you also have to remove the `#ddev-generated` at the top of that file or it will get overwritten. Do check that of course. Did your change survive restart?

But you may have to step-debug through your project and find out what it's doing, because the actual problem will likely affect production.

**Reply by DannyJoris** [reply]
Thanks for your response, @rfay ! I was able to isolate the issue to a single Drupal Paragraph. I'll put my focus toward debugging that. Thanks again!

**Reply by rfay** [reply]
Glad you got it sorted out!
